crates/pop-parachains/src/call/mod.rs

```diff
@@ -73,6 +73,47 @@ pub fn encode_call_data(
 	Ok(format!("0x{}", hex::encode(call_data)))
 }
 
+// This struct implements the [`Payload`] trait and is used to submit
+// pre-encoded SCALE call data directly, without the dynamic construction of transactions.
+struct CallData(Vec<u8>);
+
+impl Payload for CallData {
+	fn encode_call_data_to(
+		&self,
+		_: &subxt::Metadata,
+		out: &mut Vec<u8>,
+	) -> Result<(), subxt::ext::subxt_core::Error> {
+		out.extend_from_slice(&self.0);
+		Ok(())
+	}
+}
+
+/// Signs and submits a given extrinsic.
+///
+/// # Arguments
+/// * `client` - Reference to an `OnlineClient` connected to the chain.
+/// * `call_data` - SCALE encoded bytes representing the call data of the transaction.
+/// * `suri` - The secret URI (e.g., mnemonic or private key) for signing the extrinsic.
+pub async fn sign_and_submit_extrinsic_with_call_data(
+	client: OnlineClient<SubstrateConfig>,
+	call_data: &str,
```

Comment:
Minor refactor: I think we should change this type to Vec<u8> and refactor `hex::decode(call_data.trim_start_matches("0x")).map_err(|e| Error::CallDataDecodingError(e.to_string()))?;` into a function. As this is a library crate, there is the very likely case that someone has some call data bytes which they want to submit - they then need to take those bytes and convert them to a hex string first, only for them to be converted back. 

In general, strings and string handling should be handled by the UI as that is the place that is receiving user input. The UI should convert it to the type exposed by the library, in this case bytes. Another implementation may use this crate to submit an extrinsic based on bytes received from a websocket connection for example.